### PR TITLE
Add FalkorDB/RedisGraph to graph Store 

### PR DESCRIPTION
**Pull request by gkorland**
# Description

Add support for to FalkorDB/RedisGraph to Graph stores

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new notebook (that tests end-to-end)

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: llama_index/graph_stores/falkordb.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+from typing import Any, Dict, List, Optional
+
+
+from llama_index.graph_stores.types import (
+    GraphStore,
+)
+
+logger = logging.getLogger(__name__)
+
+get_query = "\
```
Comment (jerryjliu) [inline llama_index/graph_stores/falkordb.py:13]: is this used?

File: docs/community/integrations/graph_stores.md [hunk 2]
```diff
@@ -41,3 +41,17 @@ maxdepth: 1
...
+We support a `FalkorDBGraphStore` integration, for persisting graphs directly in FalkorDB! Furthermore, you can generate cypher queries and return natural language responses for your FalkorDB graphs using the `KnowledgeGraphQueryEngine`.
+
+See the associated guides below:
+
+```{toctree}
+---
+maxdepth: 1
+---
+FalkorDB Graph Store </examples/index_structs/knowledge_graph/FalkorDBGraphKGIndexDemo.ipynb>
+Knowledge Graph Query Engine </examples/query_engine/knowledge_graph_query_engine.ipynb>
```
Comment (jerryjliu) [inline docs/community/integrations/graph_stores.md:56]: i don't think this notebook has been modified, could we remove?

**Review by jerryjliu — APPROVED** [review]

**Review by wey-gu — APPROVED** [review]
Great work, thanks!

**Comment by gkorland** [comment]
> Cool, thanks for this!
> 
> Do you mind updating the changelog @gkorland ? I tried doing it myself, but I couldn't push the change to the branch
> 
> Just need to merge in the latest version of main, then add something like this image

Done.

**Comment by logan-markewich** [comment]
Thanks!! 💪💪

**Pull request merged**